cmd/snap/cmd_services.go

```diff
@@ -184,8 +185,63 @@ func (s *svcLogs) Execute(args []string) error {
 	return nil
 }
 
+type userAndScopeMixin struct {
+	System bool   `long:"system"`
+	User   bool   `long:"user"`
+	Users  string `long:"users"`
+}
+
+var userAndScopeDescs = mixinDescs{
+	// TRANSLATORS: This should not start with a lowercase letter.
+	"system": i18n.G("The operation should only affect system services."),
+	// TRANSLATORS: This should not start with a lowercase letter.
+	"user": i18n.G("The operation should only affect user services for the current user."),
+	// TRANSLATORS: This should not start with a lowercase letter.
+	"users": i18n.G("The operation should affect all user services."),
+}
+
+func (um *userAndScopeMixin) validateScopes() error {
+	switch {
+	case um.System && um.User:
+		return fmt.Errorf("--system and --user cannot be used in conjunction with each other")
+	case um.Users != "" && um.User:
+		return fmt.Errorf("--user and --users=all cannot be used in conjunction with each other")
+	case um.Users != "" && um.Users != "all":
+		return fmt.Errorf("only \"all\" is supported as a value for --users")
+	}
+	return nil
+}
+
+func (um *userAndScopeMixin) serviceScope() client.ScopeSelector {
+	switch {
+	case (um.User || um.Users != "") && !um.System:
+		return client.ScopeSelector([]string{"user"})
+	case !(um.User || um.Users != "") && um.System:
+		return client.ScopeSelector([]string{"system"})
+	}
+	return nil
+}
+
+func (um *userAndScopeMixin) serviceUsers() client.UserSelector {
+	switch {
+	case um.User:
+		return client.UserSelector{
+			Selector: client.UserSelectionSelf,
+		}
+	case um.Users == "all":
+		return client.UserSelector{
+			Selector: client.UserSelectionAll,
+		}
+	}
+	return client.UserSelector{
+		Selector: client.UserSelectionList,
+		Names:    strutil.CommaSeparatedList(um.Users),
+	}
```

Comment:
this can't really be reached now with --users=all, perhaps you can leave a comment